### PR TITLE
chore(dependencies): update spring boot

### PR DESCRIPTION
**Pull request by juanpabloprado**
Start moving incrementally towards spring boot v2.5.0, first step being bumping to v2.3.12.RELEASE

Closes https://github.com/spinnaker/spinnaker/issues/6565

`library("Spring Framework", "5.2.15.RELEASE") {`
https://github.com/spring-projects/spring-boot/blob/v2.3.12.RELEASE/spring-boot-project/spring-boot-dependencies/build.gradle#L1676

`library("Groovy", "2.5.14") {`
https://github.com/spring-projects/spring-boot/blob/v2.3.12.RELEASE/spring-boot-project/spring-boot-dependencies/build.gradle#L369

`library("jOOQ", "3.13.6") {`
https://github.com/spring-projects/spring-boot/blob/v2.3.12.RELEASE/spring-boot-project/spring-boot-dependencies/build.gradle#L894

`library("Postgresql", "42.2.20") {`
https://github.com/spring-projects/spring-boot/blob/v2.3.12.RELEASE/spring-boot-project/spring-boot-dependencies/build.gradle#L1377

`library("AssertJ", "3.16.1") {`
https://github.com/spring-projects/spring-boot/blob/v2.3.12.RELEASE/spring-boot-project/spring-boot-dependencies/build.gradle#L105

testClasses gradle task run successfully
```sh
> Task :kork-plugins:compileTestJava
> Task :kork-plugins:compileTestGroovy NO-SOURCE
> Task :kork-plugins:testClasses

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 7s
122 actionable tasks: 27 executed, 95 up-to-date
3:20:54: Execution finished 'testClasses'.
```

### CODE REVIEW

**Comment by mattgogerly** [comment]
@Mergifyio update

**Review by mattgogerly — APPROVED** [review]

**Comment by dbyron-sf** [comment]
I would have loved a chance to review this before it went in.  I have some questions about details in the PR, but I also am not sure all the code that consume kork are really ready for this change.  Kayenta may happen to work, but I think it would be more correct for it to use `enforcedPlatform` as e.g. [gate does](https://github.com/spinnaker/gate/blob/e4d271bdbdc78899186ed768a1d6c8b0a256e5d5/build.gradle#L29).

Please let PRs as fundamental as this get more feedback before merging them.

File: kork-core/kork-core.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ dependencies {
   api "org.springframework.boot:spring-boot-autoconfigure"
   api "org.springframework.boot:spring-boot-starter-aop"
   api "org.springframework.boot:spring-boot-starter-actuator"
+  api 'org.springframework.boot:spring-boot-starter-json'
```
Comment (dbyron-sf) [inline kork-core/kork-core.gradle:15]: Why is this change necessary?

File: spinnaker-dependencies/spinnaker-dependencies.gradle [hunk 2]
```diff
@@ -24,8 +24,8 @@ ext {
     spectator        : "1.0.6",
     spek             : "1.1.5",
     spek2            : "2.0.9",
-    spring           : "5.2.12.RELEASE", // this should be kept in sync with spring-boot / removed once the need for a version override is gone
-    springBoot       : "2.2.13.RELEASE",
+    spring           : "5.2.15.RELEASE", // this should be kept in sync with spring-boot / removed once the need for a version override is gone
```
Comment (dbyron-sf) [inline spinnaker-dependencies/spinnaker-dependencies.gradle:27]: If we're using the same version that spring boot brings in, why not remove this?

File: kork-core/kork-core.gradle (hunk 1 above)
Comment (juanpabloprado) [reply kork-core/kork-core.gradle:15]: classes related to jackson went missing upon upgrade like `com.fasterxml.jackson.annotation.JsonCreator;` within com/netflix/spinnaker/kork/lock/LockManager.java in kork-core
https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-json/2.3.12.RELEASE contains jackson dependencies compatible with 2.3.12.RELEASE

File: spinnaker-dependencies/spinnaker-dependencies.gradle [hunk 3]
```diff
@@ -150,7 +150,6 @@ dependencies {
     api("net.minidev:json-smart:2.4.1") // TODO: remove this with upgrade of spring-boot version to 2.5.0 or above
     api("org.apache.commons:commons-exec:1.3")
     api("org.apache.commons:commons-lang3:3.9")
-    api("org.assertj:assertj-core:3.15.0")
```
[the remark this answers is not in the document]
Comment (juanpabloprado) [reply spinnaker-dependencies/spinnaker-dependencies.gradle:153]: v2.3.12.RELEASE comes with [AssertJ](https://mvnrepository.com/artifact/org.assertj/assertj-core) 3.16.1 which is one minor version away from 3.15.0, release order being 3.15.0, 3.16.0, 3.16.1

File: spinnaker-dependencies/spinnaker-dependencies.gradle (hunk 2 above)
Comment (juanpabloprado) [reply spinnaker-dependencies/spinnaker-dependencies.gradle:27]: not 100% about removing it because of the comment about keeping it in sync with spring-boot

Comment (dbyron-sf) [reply spinnaker-dependencies/spinnaker-dependencies.gradle:27]: If you're not sure then I agree keeping it is thing to do....though the comment does also say "removed once the need for a version override is gone" ... I guess the work to truly verify this remains...though this kind of PR feels like the right time to do it.
